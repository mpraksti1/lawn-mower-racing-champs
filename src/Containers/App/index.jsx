import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Provider as AlertProvider } from 'react-alert';
import AlertTemplate from 'react-alert-template-basic';
import AppHeader from '../../Components/AppHeader';
import Auth from '../Auth';
import Site from '../Site';
import StyledDiv from './styles';
import 'whatwg-fetch';

const options = {
  position: 'top center',
  timeout: 5000,
  offset: '30px',
  transition: 'scale',
};

const App = () => (
  <AlertProvider template={AlertTemplate} {...options}>
    <StyledDiv>
      <AppHeader />
      <main>
        <Switch>
          <Route path="/auth" component={Auth} />
          <Route path="/" component={Site} />
        </Switch>
      </main>
    </StyledDiv>
  </AlertProvider>
);

export default App;
