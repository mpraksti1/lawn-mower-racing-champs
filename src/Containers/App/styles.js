import styled from 'styled-components';

const StyledDiv = styled.div`
  font-size: 100%;
  font-family: 'Gotham';
  background: #333;
`;

StyledDiv.displayName = 'AppStyled';

export default StyledDiv;
