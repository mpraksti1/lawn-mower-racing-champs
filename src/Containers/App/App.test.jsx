import React from 'react';
import { shallow } from 'enzyme';
import App from '.';

describe('App component', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
