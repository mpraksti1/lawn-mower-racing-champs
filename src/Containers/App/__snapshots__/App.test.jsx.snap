// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<Provider
  offset="30px"
  position="top center"
  template={[Function]}
  timeout={5000}
  transition="scale"
  type="info"
  zIndex={100}
>
  <AppStyled>
    <withRouter(Connect(AppHeader))
      checkAuthStatus={[Function]}
    />
    <main>
      <Switch>
        <Route
          component={[Function]}
          path="/auth"
        />
        <Route
          path="/"
          render={[Function]}
        />
      </Switch>
    </main>
  </AppStyled>
</Provider>
`;
